### ISSUE TITLE
Deploy VPC resources with non crossplane VPC

### ISSUE THREAD

**Issue by afreyermuth98** [crossplane-contrib/provider-aws#2101]
### What happened?

I wanted to deploy a security group in my existing AWS VPC.
But it looks like we need to have created the VPC using crossplane to deploy resources in int which is kinda annoying.
Is there a way to do it or it's mandatory to have the VPC created with crossplane ?

```
cannot resolve references: mg.Spec.ForProvider.VPCID: cannot get
referenced resource: VPC.ec2.aws.crossplane.io "vpc-XXX"
not found
```

### How can we reproduce it?

Create a ec2 security group with crossplane that references a VPC non created with crossplane

### What environment did it happen in?
Crossplane version: 1.17.0

* Cloud provider or hardware configuration: AWS
* Kubernetes version (use `kubectl version`) : 1.29
* Kubernetes distribution (e.g. Tectonic, GKE, OpenShift): EKS

**Comment by bobh66** [comment]
You can specify the `spec.forProvider.vpcId` explicitly in the `SecurityGroup` resource.  If you use `spec.forProvider.vpcRef.name` then it will look for a corresponding Crossplane resource.

**Comment by bobh66** [comment]
Yes - as long as there is no Managed Resource that controls the VPC it will remain untouched.

**Comment by zrml** [comment]
@afreyermuth98 did you resolve the issue? I see a similar one with igw and ngw... thanks
